Golang no longer a prequisite, got rid of globally required imports

Mingling94 opened the pull request:
Various changes that _seem_ unrelated but actually are. Explanation in chronological order:
- Fixed `.finalizeRelease.sh` as it wasn't putting the correct Sia release into each Sia-UI release before
  - This took a bunch of inspiration from David's previous release.sh code
- From there I saw that Golang wasn't needed for anything else besides `npm run sia-import` and so I added `justGetSia.js` to download sia on post-install instead of relying on the user to have a `$GOPATH/bin/siad`
- Fixed bug with how `siad.configure` was being called. Irrelevant with #155 but twas a quick fix.
- Because jshint was complaining about the `justGetSia.js` redefinition of the globally require 'path' node library, I got rid of global imports on the general UI level. They seemed like more trouble than they were worth anyway.

Mingling94 committed:
Fixed finalizeRelease script

lukechampine commented on .finalizeRelease.sh:
```diff
@@ -1,35 +1,84 @@
+	local extension=".zip"
+	if [ "${1:0:5}" == "linux" ]; then
+		extension=".tar.gz"
+	fi
```
declaring the extension here seems unnecessary since you branch on arch anyway

Mingling94 replied:
It makes it so I can do the following series:

```
local sia_archive="${sia_folder}${extension}"
local sia_url="https://github.com/NebulousLabs/Sia/releases/download/${VERSION}/${sia_archive}"
```

to get the one-line of wget (instead of 3):

``` bash
wget -q $sia_url
```

Mingling94 committed:
`npm install` downloads Sia on postinstall script

This, with the previous commit, means a golang setup is no longer a
prerequisite to running the UI

Mingling94 committed:
Was using siad.configure wrongly

Mingling94 committed:
Got rid of globally imported libraries

Realized the .jshintrc management wasn't worth the infinitesimal
performance boost it offered.

Also importing needed libraries per file
makes the code more readable.

Also also the performance boost of not having to require per file or per
usage was counterbalanced against the memory usage anyway.

Mingling94 commented:
@DavidVorick @lukechampine @VoidingWarranties boop
